WIP: Implemented RGPM Interface and other fixes

oparkins opened the pull request:
Fixes the webpack bundle. Do this pull request after #17

oparkins committed:
Removed all node_modules from being using in the webpack version. This closes #16

oparkins committed:
Fixed spelling mistake in Makefile

oparkins commented:
Ignore the failing builds right now... There appears to be an issue with Travis-CI Xenial image. I put in a comment at https://travis-ci.community/t/xenial-kernel-does-not-appear-to-have-sandbox-capabilities/5370.

oparkins committed:
Updated travis ci to use trusty

oparkins committed:
Working on fixing travis ci

oparkins committed:
feat: Added initial CRUD interface to the RGPM file. This closes issue #23

oparkins committed:
feat: Added rgpm/storage-integrations as a dependency

Since the storage-integrations repo has a npm package on github, yarn has been configured to use the github repo.

irremotus commented:
It looks like these changes may need to be rebased onto master.  It seems like there are duplicate changes here as in #17.

dependabot[bot] committed:
Bump handlebars from 4.2.0 to 4.5.3

Bumps [handlebars](https://github.com/wycats/handlebars.js) from 4.2.0 to 4.5.3.
- [Release notes](https://github.com/wycats/handlebars.js/releases)
- [Changelog](https://github.com/wycats/handlebars.js/blob/master/release-notes.md)
- [Commits](https://github.com/wycats/handlebars.js/compare/v4.2.0...v4.5.3)

Signed-off-by: dependabot[bot] <support@github.com>

oparkins committed:
Merge remote-tracking branch 'origin/dependabot/npm_and_yarn/handlebars-4.5.3' into issue/#16

oparkins committed:
chore: Updated all dependencies

Updated dependencies using yarn upgrade --latest. The tests passed. This was to fix some security issues that were mentioned by GitHub

oparkins committed:
feat: Implemented null_concat function

oparkins committed:
test: moved null_concat testing to cryptoFactory tests

I moved the tests to where the actual crypto lib will get tested. This also makes sure that the methods are accessible in the library itself and that the functions work across different javascript platforms. (even though that is technically not necessary)

oparkins committed:
chore: Updated dependencies and configurations for CRUD operation implementations

Had to add storage-integrations as a dependency. Ran into an issue where
the webpack version upgrade failed testing. I had to change the webpack
config to get it to work again.

oparkins committed:
feat: Added storage decision in RGPM class

When initializing the RGPM class, it will now determine which storage location should be used. The default is local storage and when it is the first time the class has ever been initialized, it will use the local storage as well.

oparkins committed:
Added mapHashToPass and start of InitPass to rgpm

oparkins committed:
Updated createRecord method to start reflecting required data for record creation

oparkins committed:
Implemented more of initPass method

oparkins committed:
refactor: Moved verify method from crypto to rgpm

The verify method doesn't deal with crypto, but instead uses it. This is like the other functions inside of rgpm. In addition, by moving the method to rgpm, it allows for the test cases to be simplier and cleaner by not duplicated password specifications through the test files

oparkins committed:
feat: Implemented verify function

oparkins committed:
refactor: Changed for loop to make the body of the loop more readable

oparkins committed:
feat: Implemented initPass method

irremotus commented on src/crypto.js:
```diff
@@ -23,12 +23,21 @@ class Crypto {
-    verify() { 
-        throw new NotImplementedError("verify has not been implemented yet");
+        // Remove the last null byte from the string
+        return result.substring(0, result.length - 1);
```
Is there a reason Array.join wasn't used here?

oparkins replied:
I have no excuse.... I'm not sure why I didn't use that call - I'm losing my mind.

I'll fix that asap

oparkins committed:
refactor: changed the return types of the crytpo functions to better suit the algorithms that use them

Instead of converting everything to hex each time we needed to use it, I made the crypto functions able to use the output of the other crypto functions. This was a problem that was seen during testing of initPass in rgpm.js.

irremotus commented on src/rgpm.js:
```diff
@@ -2,18 +2,133 @@
     {
-        let Crypto = CryptoFactory.selectCrypto();
+        const Crypto = CryptoFactory.selectCrypto();
+        let Storage =  this.getStorage();
```
Should these be instance variables instead?

oparkins replied:
Yup. Caught that earlier. Not sure why I did that originally, but when I was testing the code the tests threw that error.

oparkins committed:
fix: Updated and tested initPass.

irremotus commented on test/cryptoFactory.test.js:
```diff
@@ -80,4 +81,25 @@ describe("cryptoFactory and crypto methods", () => {
+        [[], ""]
+    ])("null_concat should produce correct output", (items, output) => {
+        
+        test("on nodejs", () => {
+            expect(crypto.null_concat.apply(this, items)).toEqual(output);
+        });
+
+        test("on browser", async () => {
+            const result = await page.evaluate(async (items) => {    
+                const cryptoLib = app.selectCrypto();
+                return cryptoLib.null_concat.apply(this, items);
+            }, items);
+            expect(result).toEqual(output);
+        });
```
Why can these two tests not be more similar?

oparkins replied:
Because of the architecture that the browser tests use, puppeteer. I can't select the crypto version to use outside of the page evaluate function. As for where the `expect(result).toEqual(output)` line, I believe it created errors when it is inside the page evaluation call. I don't remember exactly though... I can put it in the list to look at. Does that answer your question?

irremotus replied:
Yes, thanks

irremotus commented on test/crypto.test.js:
```diff
@@ -20,13 +20,11 @@ describe("crypto methods", () => {
         "digest",
         "hmac",
         "null_concat",
-        "verify",
```
Why is verify removed?

oparkins replied:
Because verify doesn't exist inside the crypto.js file any more. I moved it to RGPM. Since it deals more with the passwords and not the crypto, I didn't think it was too relevant in the crypto file.

irremotus commented:
Yes, thanks

________________________________
From: Owen Parkins <notifications@github.com>
Sent: Monday, December 30, 2019 1:58:12 PM
To: rgpm/core <core@noreply.github.com>
Cc: Kevin Schmittle <kevin@schmittle.net>; Review requested <review_requested@noreply.github.com>
Subject: Re: [rgpm/core] WIP: Implemented RGPM Interface and other fixes (#20)


@oparkins commented on this pull request.

________________________________

In test/cryptoFactory.test.js<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frgpm%2Fcore%2Fpull%2F20%23discussion_r362094685&data=02%7C01%7C%7Cbe0e2981ecac45c59a7408d78d6afc46%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637133362939147172&sdata=GQvSpwXKyJGdS2Kg0XMOKV5J5Buzb4vBFtxFzd7oKCc%3D&reserved=0>:

> +        test("on nodejs", () => {
+            expect(crypto.null_concat.apply(this, items)).toEqual(output);
+        });
+
+        test("on browser", async () => {
+            const result = await page.evaluate(async (items) => {
+                const cryptoLib = app.selectCrypto();
+                return cryptoLib.null_concat.apply(this, items);
+            }, items);
+            expect(result).toEqual(output);
+        });


Because of the architecture that the browser tests use, puppeteer. I can't select the crypto version to use outside of the page evaluate function. As for where the expect(result).toEqual(output) line, I believe it created errors when it is inside the page evaluation call. I don't remember exactly though... I can put it in the list to look at. Does that answer your question?

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frgpm%2Fcore%2Fpull%2F20%3Femail_source%3Dnotifications%26email_token%3DAAUBBA7IQL37TBOENGCWUBDQ3JOGJA5CNFSM4I6ZZDK2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCQM5IJY%23discussion_r362094685&data=02%7C01%7C%7Cbe0e2981ecac45c59a7408d78d6afc46%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637133362939157177&sdata=hjHYoV159D1DCpcXjsj9VJvBn0PGB6YBIsBOqBY0t1k%3D&reserved=0>, or unsubscribe<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAUBBA4KKRQIMM42HJLZXITQ3JOGJANCNFSM4I6ZZDKQ&data=02%7C01%7C%7Cbe0e2981ecac45c59a7408d78d6afc46%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637133362939167188&sdata=yhRVSFbvNzriQQPV86e238La78GtCdmUCT0m7hGlBUk%3D&reserved=0>.

oparkins committed:
refactor: Changed null_concat to use Array.join method

oparkins committed:
Added CRUD operations to records.

Implemented the record CRUD operations which also includes test cases. Added the genPass algorithm as well. This will mark the end of the major development of the core, and where the GUI work starts

oparkins committed:
Updated package name for publishing reasons

oparkins committed:
feat: Added ability to list all records that have been created

oparkins committed:
v1.1.0

oparkins committed:
refactor: added option to add iter_t value when creating record

oparkins committed:
v1.2.0

oparkins committed:
refactor: fixed spelling mistake with genPass testing method

oparkins committed:
fix: allowed for the initPass algorithm to not overwrite the revision count if it is different than 1

oparkins committed:
feat: added function to update record to next revision

oparkins committed:
fix: Fixed the dockerfile build errors due to base image changing included packages

oparkins committed:
feat: added functions to deal with previous passwords and changing revisions

oparkins committed:
v1.3.0

oparkins committed:
feat: added ability to calculate the optimal iter_t value
